node/endpoint-handler.js

```diff
@@ -72,16 +73,30 @@ TChannelEndpointHandler.prototype.register = function register(name, handler) {
 
 TChannelEndpointHandler.prototype.handleRequest = function handleRequest(req, res) {
     var self = this;
-    var name = String(req.arg1);
-    var handler = self.endpoints[name];
-    if (!handler) {
-        res.sendNotOk(null, NoSuchEndpointError({
-            service: self.serviceName,
-            endpoint: name
-        }).message);
-        return;
-    }
-    handler(req, res);
+    // TODO: waterfall
+    req.arg1.onValueReady(function arg1Ready(err, arg1) {
+        if (err) throw err; // TODO: protocol error, respond with error frame
```

Comment:
This is scary to add to master.

Can we make a `logger.error()` call and a `res.writeErrorFrame('BadRequest', 'you done messed up');`